instaclustr/api_client.go

```diff
@@ -390,3 +390,23 @@ func (c *APIClient) UpdateBundleUser(clusterID string, bundle string, data []byt
 	}
 	return nil
 }
+
+func (c *APIClient) UpdateEndpointServicePrincipals(clusterDataCentreID string, data []byte) error {
```

Comment:
Can we have a unit test to cover this func? But I saw Arjun have already approved it, so I guess it's up to you now. Not very necessary.

Reply:
Makes sense I will include a unit test.